src/components/action-popover/action-popover.pw.tsx

```diff
@@ -185,6 +185,29 @@ test.describe("check functionality for ActionPopover component", () => {
     await expect(focusedElement).toContainText("Sub Menu 2");
   });
 
+  test("should not scroll to the top of the document when first menu item is focused and component is rendered in a tabular parent", async ({
+    mount,
+    page,
+  }) => {
+    await page.setViewportSize({
+      width: 700,
+      height: 300,
+    });
+    await mount(<ActionPopoverComponentInFlatTable />);
+
+    await page.evaluate(() => window.scrollTo(0, 1000));
+    const actionPopoverButtonElement = await actionPopoverButton(page).nth(0);
```

Comment:
**suggestion/nitpick(non-blocking):** Might be worth double checking if all these `awaits`s are needed in the `const`s. If the tests don't work without them, that'll probably give you your answer.